fiftyone/core/odm/database.py

```diff
@@ -172,6 +172,8 @@ def establish_db_conn(config):
             % (config.type, foc.CLIENT_TYPE)
         )
 
+    _delete_non_persistent_datasets_if_necessary()
```

Comment:
Important to do this last because https://github.com/voxel51/fiftyone/pull/1692 added a check to this method that may reject certain client connections